pkg/pool-gyro/contracts/GyroECLPPool.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: LicenseRef-Gyro-1.0
+// for information on licensing please see the README in the GitHub repository
+// <https://github.com/gyrostable/concentrated-lps>.
+
+pragma solidity ^0.8.24;
+
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import { IBasePool } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePool.sol";
+import { IGyroECLPPool } from "@balancer-labs/v3-interfaces/contracts/pool-gyro/IGyroECLPPool.sol";
+import { ISwapFeePercentageBounds } from "@balancer-labs/v3-interfaces/contracts/vault/ISwapFeePercentageBounds.sol";
+import {
+    IUnbalancedLiquidityInvariantRatioBounds
+} from "@balancer-labs/v3-interfaces/contracts/vault/IUnbalancedLiquidityInvariantRatioBounds.sol";
+import { PoolSwapParams, Rounding, SwapKind } from "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+
+import { BalancerPoolToken } from "@balancer-labs/v3-vault/contracts/BalancerPoolToken.sol";
+
+import { GyroECLPMath } from "./lib/GyroECLPMath.sol";
+
+/**
+ * @notice Standard Gyro E-CLP Pool, with fixed E-CLP parameters.
+ * @dev Gyroscope's E-CLPs are AMMs where trading takes place along part of an ellipse curve. A given E-CLP is
+ * parameterized by the pricing range [α,β], the inclination angle `phi` and stretching parameter `lambda`. For more
+ * information, please refer to https://docs.gyro.finance/gyroscope-protocol/concentrated-liquidity-pools/e-clps.
+ */
+contract GyroECLPPool is IGyroECLPPool, BalancerPoolToken {
+    using FixedPoint for uint256;
+    using SafeCast for *;
+
+    /// @dev Parameters of the ECLP pool
+    int256 internal immutable _paramsAlpha;
+    int256 internal immutable _paramsBeta;
+    int256 internal immutable _paramsC;
+    int256 internal immutable _paramsS;
+    int256 internal immutable _paramsLambda;
+
+    /**
+     * @dev Derived Parameters of the E-CLP pool, calculated off-chain based on the parameters above. 38 decimals
+     * precision.
+     */
+    int256 internal immutable _tauAlphaX;
+    int256 internal immutable _tauAlphaY;
+    int256 internal immutable _tauBetaX;
+    int256 internal immutable _tauBetaY;
+    int256 internal immutable _u;
+    int256 internal immutable _v;
+    int256 internal immutable _w;
+    int256 internal immutable _z;
+    int256 internal immutable _dSq;
+
+    bytes32 private constant _POOL_TYPE = "ECLP";
+
+    constructor(GyroECLPPoolParams memory params, IVault vault) BalancerPoolToken(vault, params.name, params.symbol) {
```

Comment:
I believe that's used as a replacement for poolCreator in v2. I mean, the missing address stored protocol and gyro fees and was used to distribute fees on exit. I'll keep it without it, since V3 has its own mechanism of doing that, but I'll ask in the Gyro group if that's ok.